api-2.1/src/main/java/org/openmrs/module/fhir2/api/translators/impl/ObservationInterpretationTranslatorImpl_2_1.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.translators.impl;
+
+import org.hl7.fhir.r4.model.CodeableConcept;
+import org.openmrs.Obs;
+import org.openmrs.annotation.OpenmrsProfile;
+import org.openmrs.module.fhir2.api.translators.ObservationInterpretationTranslator;
+import org.springframework.context.annotation.Primary;
+import org.springframework.stereotype.Component;
+
+@Primary
+@Component
+@OpenmrsProfile(openmrsPlatformVersion = "2.1.* - 2.3.*")
+public class ObservationInterpretationTranslatorImpl_2_1 implements ObservationInterpretationTranslator {
+	
+	@Override
+	public CodeableConcept toFhirResource(Obs obs) {
+		CodeableConcept interpretation = new CodeableConcept();
+		if (obs.getInterpretation() != null) {
+			interpretation.setText(obs.getInterpretation().toString());
```

Comment:
So what we actually want to do here is to map not to text but to the coding using the code set defined [here](https://www.hl7.org/fhir/valueset-observation-interpretation.html). So, for example, `Obs.Interpretation.NORMAL` should map to a structure that looks like this:

```json
"interpretation": {
	"coding": [
		{
			"system": "http://hl7.org/fhir/ValueSet/observation-interpretation",
			"code": "N",
			"display": "Normal
		}
	],
	"text": "Normal"
}
```

Code-wise this looks something like (with the constants appropriately extracted out):

```java
switch(obs.getInterpretation()) {
    case NORMAL:
		interpretation.addCoding().setSystem("http://hl7.org/fhir/ValueSet/observation-interpretation").setCode("N").setDisplay("Normal");
		interpretation.setText("Normal");
	...
}
```

The mappings should be something like:

|OpenMRS Type|FHIR Interpretation|
|-------|------|
|`NORMAL`|`N`|
|`ABNORMAL`|`A`|
|`CRITICALLY_ABNORMAL`|`AA`|
|`CRITICALLY_HIGH`|`HH`|
|`CRITICALLY_LOW`|`LL`|
|`HIGH`|`H`|
|`LOW`|`L`|
|`OFF_SCALE_LOW`|`<`|
|`OFF_SCALE_HI`|`>`|
|`SIGNIFICANT_CHANGE_DOWN`|`D`|
|`SIGNIFICANT_CHANGE_UP`|`U`|
|`RESISTANT`|`R`|
|`SUSCEPTIBLE`|`S`|
|`INTERMEDIATE`|`I`|
|`POSITIVE`|`POS`|
|`NEGATIVE`|`NEG`|

The display values should be those in the valueset I linked above.

Some special handling is needed for the `VERY_SUSCEPTIBLE` interpretation, since this isn't supported by the valueset. For this case, it might make sense to add the coding for susceptible and then create an additional code with the system `https://fhir.openmrs.org/ext/obs/interpretations`, the code `VS`, and the display "Very Susceptible", which should also be the reported text.

Reply:
This is noted, working on it..